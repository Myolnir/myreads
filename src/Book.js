import React,  { Component } from 'react'
import PropTypes from 'prop-types'
import BookSelector from './booksSelector'


class Book extends Component {

  static propTypes = {
    book: PropTypes.array.isRequired
  }

  render() {
    debugger;
    const { book } = this.props
    return (
      <li key={book.id}>
        <div className="book">
          <div className="book-top">
            <div className="book-cover"
              style={
                  {
                    width: 128,
                    height: 193,
                    backgroundImage: `url("${book.imageLinks.thumbnail}")`
                  }
                }>
              </div>
            <BookSelector />
          </div>
          <div className="book-title">{book.title}</div>
          <div className="book-authors">{book.author}</div>
        </div>
      </li>
    )
  }
}

export default Book
