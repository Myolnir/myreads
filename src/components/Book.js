import React,  { Component } from 'react'
import PropTypes from 'prop-types'
import BookSelector from './booksSelector'


class Book extends Component {

  static propTypes = {
    book: PropTypes.object.isRequired,
    moveToShelf: PropTypes.func.isRequired
  }

  render() {
    const { book, moveToShelf } = this.props
    return (
      <li key={book.id}>
        <div className="book">
          <div className="book-top">
            <div
              className="book-cover"
              style={
                {
                  width: 128,
                  height: 193,
                  backgroundImage:
                  `url("${book.imageLinks && book.imageLinks.thumbnail
                  }")`
                }
                }>
              </div>
            <BookSelector
              book={book}
              moveToShelf={moveToShelf}
            />
          </div>
          <div className="book-title">{book.title}</div>
          <div className="book-authors">{book.author}</div>
        </div>
      </li>
    )
  }
}

export default Book
