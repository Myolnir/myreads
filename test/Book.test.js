import React from 'react';
import { expect } from 'chai';
import { shallow, mount, render } from 'enzyme';
import Book from '../src/components/Book';

describe("<Book />", () => {
  it("contains spec with an expectation", ()  => {
    const book = {};
    const moveToShelf = function(){}
    const bookWrapper = mount(
      <Book
        book={book}
        moveToShelf={moveToShelf}
      />
    )
    expect(bookWrapper.contains(book)).to.equal(true);
  });

});
