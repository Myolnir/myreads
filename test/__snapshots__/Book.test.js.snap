exports[`<Book /> Should render the component 1`] = `"<li><div class=\"book\"><div class=\"book-top\"><div class=\"book-cover\" style=\"width: 128px; height: 193px; background-image: url(http://localhost/test.jpg);\"></div><div class=\"book-shelf-changer\"><select><option value=\"none\" disabled=\"\">Move to...</option><option value=\"currentlyReading\">Currently Reading</option><option value=\"wantToRead\">Want to Read</option><option value=\"read\">Read</option><option value=\"none\">None</option></select></div></div><div class=\"book-title\">test title</div><div class=\"book-authors\">author title</div></div></li>"`;
